# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: servo/components/style/properties/declaration_block.rs
# Commit: 6f62586a139b
# Full Hash: 6f62586a139b1fcc4817dddd924400958d89a6bb
# Author: Zach Hoffman <zach@zrhoffman.net>
# Date: 2023-12-18 17:17:57
# Regressor Bug: 1846516
# File Overlap Count: 2
# Description:
#   Bug 1846516 - [css-properties-values-api] Animate custom properties in a discrete way. r=firefox-style-system-reviewers,emilio
#   
#   This patch also updates the bug ID for a FIXME leftover from bug 1840478
#   to bug 1869476, since the same FIXME is added in D190758.
#   
# ==============================================================================

diff -r 729abedf9bda -r 6f62586a139b servo/components/style/properties/declaration_block.rs
--- a/servo/components/style/properties/declaration_block.rs	Mon Dec 18 09:24:08 2023 +0000
+++ b/servo/components/style/properties/declaration_block.rs	Mon Dec 18 09:24:08 2023 +0000
@@ -174,10 +174,25 @@
 
     /// Returns whether this set contains all longhands in the specified set.
     #[inline]
-    pub fn contains_all(&self, longhands: &LonghandIdSet) -> bool {
+    pub fn contains_all_longhands(&self, longhands: &LonghandIdSet) -> bool {
         self.longhands.contains_all(longhands)
     }
 
+    /// Returns whether this set contains all properties in the specified set.
+    #[inline]
+    pub fn contains_all(&self, properties: &PropertyDeclarationIdSet) -> bool {
+        if !self.longhands.contains_all(&properties.longhands) {
+            return false;
+        }
+        if properties.custom.len() > self.custom.len() {
+            return false;
+        }
+        properties
+            .custom
+            .iter()
+            .all(|item| self.custom.contains(item))
+    }
+
     /// Iterate over the current property declaration id set.
     pub fn iter(&self) -> PropertyDeclarationIdSetIterator {
         PropertyDeclarationIdSetIterator {
@@ -1102,7 +1117,7 @@
                 //     If all properties that map to shorthand are not present
                 //     in longhands, continue with the steps labeled shorthand
                 //     loop.
-                if !self.property_ids.contains_all(&longhands) {
+                if !self.property_ids.contains_all_longhands(&longhands) {
                     continue;
                 }
 